fuse/dl/README.md

```diff
@@ -0,0 +1,75 @@
+                                       validation_metrics=validation_metrics,
+                                       best_epoch_source=best_epoch_source,
+                                       optimizers_and_lr_schs=optimizers_and_lr_schs)
+```
+where the user is only responsible to supply the required arguments, including the model, losses, train and validation metrics, all of which are in fuse style (classes implemented or supported in FuseMedML).
+
+    
+### 2. Custom use (suitable for any use case)
+(see [MNIST example](../../examples/fuse_examples/imaging/classification/mnist/run_mnist_custom_pl_imp.py))  
+
+In this more flexible and custom mode, the user is responsible to implement their own `LightningModule` tailored to their project, with all standard methods required by PyTorch Lightning such as `forward`, `training_step` etc'.  
+Once such a custom module exists, it is instantiated with a user defined set of arguments:
+```
+pl_module = CustomLightningModule(**custom_args)
+```  
+
+Note: previously, a dedicated PyTorch based [manager](managers) was used which is now deprecated. Some of the implemented [examples](../../examples) still use it but we plan to port them to use the lightning manager.
+
+## losses
+This module implements basic FuseMedML loss classes as well as a few specific loss function implementations.
+The basic classes are:  
+1. `LossBase` - A generic template class that can be used for implementing any loss function. All it does is receives a `batch_dict` and returns a loss tensor.
+2. `LossWarmUp` - A wrapper class that gets an existing loss function and zeros its value for a given number of iterations. This is useful in a multiple loss function scenario, where one loss should be stabilized before the other one comes into play.
+3. `LossDefault` - Default loss implementation of FuseMedML. It is still a generic class in the sense that is can operate on any actual loss function implementation. In addition, it can receive a scalar multiplier value, and a custom preprocessing function that operated on the `batch_dict` before the loss operates on it.  
+[The MNIST example uses `LossDefault`](../../examples/fuse_examples/imaging/classification/mnist/run_mnist.py)
+
+The specific losses currently implemented are suitable for classification and dense segmentation tasks:  
+1. `LossSegmentationCrossEntropy` - Cross-entropy loss with optional class weights and resizing of the ground truth map.
+2. `BinaryDiceLoss` - Dice loss with optional class weights, `batch_dict` preprocessing and ground truth map resizing. 
+3. `FocalLoss` - Focal loss with configurable hyperparameters and optional scalar weight, `batch_dict` preprocessing and ground truth map resizing.  
+
+Note again, that while these are specific loss functions that are implemented, it is possible to use any loss, whether custom, or one that already exists in PyTorch, and pass it to the `LossDefault` class for use in FuseMedML.
+
+## models
```

Comment:
Maybe start with that the FuseMedML style model is almost similar to PyTorch.
The only differences are that it gets as input batch_dict and returns batch_dict["model"] (model outputs)

Reply:
done